### PR TITLE
Fix one failing E2E test on Firefox, skip the other one

### LINKED ISSUE #1801: debt: 2 failing Firefox playwright E2E tests after npm packages
Curiously, after [updating our npm dependencies](https://github.com/sillsdev/web-languageforge/pull/1799) there are two Firefox playwright E2E tests that consistently [fail](https://github.com/sillsdev/web-languageforge/actions/runs/8890540339/job/24415644965):

> 2 failed
> [firefox] › editor/editor.entries.spec.ts:207:7 › Editor entries › Create new word, modify new word, autosaves changes, new word visible in editor and list
> [firefox] › editor/editor.pictures.spec.ts:53:7 › Editor pictures › Showing and hiding captions
> 17 passed (3.1m)

### PR DESCRIPTION
**Pull request by rmunn**
## Description

We currently have two E2E tests that fail on Firefox while they pass on Chrome.

One of them appears to be caused by `reload()` not triggering our page autosave when it happens immediately after a `locator.fill()`. I suspect that the `reload()` isn't triggering the `blur` event which would allow our autosave to trigger, but only on Firefox (for reasons that would take so long to determine that I figure it's not worth spending the time). I solved this one by clicking on another entry before running the `reload()`, which forces the just-entered data to autosave and thus the rest of the test passes as it should.

The other test appears to be running into [a Playwright bug with `locator.fill()` on Firefox](https://www.github.com/microsoft/playwright/issues/29614), where the locator thinks the input field (or textarea in this case) is not editable. This happens even if you put `expect(...).toBeEditable()` immediately before the `.fill()`. The `expect` passes, but then the locator waits forever (or rather, until the test times out) for the text area to be editable, for some reason not realizing that it is editable. Since this appears to be caused by a Playwright and/or Firefox bug rather than by our code, and the test passes in Chromium, I decided to simply have this test skipped on Firefox until such time as the `locator.fill()` bug gets resolved.

## Checklist

- [X] I have labeled my PR with: bug, feature, engineering, security fix or testing
- [X] I have performed a self-review of my own code
- [X] I have reviewed the title & description of this PR which I will use as the squashed PR commit message
- [X] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have enabled auto-merge (optional)

## Testing

- Run `npm run test-e2e`; all E2E tests should now pass.

### CODE REVIEW

File: test/e2e/tests/editor/editor.pictures.spec.ts [hunk 1]
```diff
@@ -51,8 +51,6 @@ test.describe('Editor pictures', () => {
   });
 
   test('Showing and hiding captions', async ({ managerTab, browserName }) => {
-    test.slow(browserName === 'firefox');
```
Comment (myieye) [inline test/e2e/tests/editor/editor.pictures.spec.ts:54]: Why did you remove this? We really don't want to have to debug this again, because it sometimes takes too long on Firefox. I'd say just leave it unless you're confident it's really not necessary.

File: test/e2e/tests/editor/editor.pictures.spec.ts [hunk 2]
```diff
@@ -76,7 +74,11 @@ test.describe('Editor pictures', () => {
       await expect(caption).toBeVisible(); // it disappears immediately which could be annoying
       await caption.fill('');
       await expect(caption).not.toBeVisible(); // it disappears immediately which could be annoying
+      // Navigate away to force changes to be saved
```
Comment (myieye) [inline test/e2e/tests/editor/editor.pictures.spec.ts:77]: Did you try just doing something like `caption.blur()`? (I'm not sure if that method exists, but surely there's an API somewhere that'll do it)

**Review by megahirt — COMMENTED** [review]
This is great detective work, Robin!  Once you've resolved @myieye 's comments I'd like to get this merged so that future PRs can benefit from the green this will bring back.

File: test/e2e/tests/editor/editor.pictures.spec.ts (hunk 1 above)
Comment (rmunn) [reply test/e2e/tests/editor/editor.pictures.spec.ts:54]: Restored in commit dba024ceb. I removed it because it didn't seem necessary in my test runs on my local machine, but it's true that GitHub Actions runners are often slower, so I put it back.

File: test/e2e/tests/editor/editor.pictures.spec.ts (hunk 2 above)
Comment (rmunn) [reply test/e2e/tests/editor/editor.pictures.spec.ts:78]: There is a `locator.blur()` method, and I didn't think to try that. I'll try it and see if that's enough.

Comment (rmunn) [reply test/e2e/tests/editor/editor.pictures.spec.ts:78]: Interestingly, doing `caption.blur()` immediately after `caption.fill('')` caused the test to fail because "Error: locator.blur: Target page, context or browser has been closed". Which I think is incorrect; I think the actual problem is that as soon as you empty the caption, it gets hidden, so `.blur()` suddenly can't find the locator. I'll try again with `if (caption.isVisible()) { caption.blur() }` and see what that does.

Comment (rmunn) [reply test/e2e/tests/editor/editor.pictures.spec.ts:78]: Well, `caption.blur()` didn't help. Probably because the caption immediately goes invisible and so the Playwright locator can't call `.blur()` on it, and so Firefox isn't sending the blur event. I'll try one more thing: calling .focus() on a different locator, which should cause the blur event to be fired on the caption. If that fails I'll stick with the code as-is.

Comment (rmunn) [reply test/e2e/tests/editor/editor.pictures.spec.ts:78]: Calling `.focus()` on the lexeme field didn't work on Firefox either. I'm going to leave this as-is, because I have not gotten the simpler alternatives to work and this is simple enough, all things considered.

Comment (rmunn) [reply test/e2e/tests/editor/editor.pictures.spec.ts:78]: @myieye - Please feel free to resolve this conversation if you're satisfied. I'm leaving it open so it's easier for you to find when you do a re-review.

**Review by myieye — APPROVED** [review]
